sdk/eventgrid/azure-eventgrid/tests/conftest.py

```diff
@@ -23,32 +23,49 @@
 # IN THE SOFTWARE.
 #
 # --------------------------------------------------------------------------
-import platform
-import sys
+import os
 import pytest
-from devtools_testutils import test_proxy
+from devtools_testutils import add_general_string_sanitizer, test_proxy, add_body_key_sanitizer, add_header_regex_sanitizer, add_oauth_response_sanitizer, add_body_regex_sanitizer
 from devtools_testutils.sanitizers import (
     add_remove_header_sanitizer,
     add_general_regex_sanitizer,
     set_custom_default_matcher,
 )
 
-# Ignore async tests for Python < 3.5
-collect_ignore_glob = []
-if sys.version_info < (3, 5):
-    collect_ignore_glob.append("*_async.py")
-    collect_ignore_glob.append("test_cncf*")
-
-
 @pytest.fixture(scope="session", autouse=True)
-def add_aeg_sanitizer(test_proxy):
+def add_sanitizers(test_proxy):
     # this can be reverted to set_bodiless_matcher() after tests are re-recorded and don't contain these headers
     set_custom_default_matcher(
         compare_bodies=False,
         excluded_headers="Authorization,Content-Length,x-ms-client-request-id,x-ms-request-id",
     )
-    add_remove_header_sanitizer(headers="aeg-sas-key, aeg-sas-token")
+    add_remove_header_sanitizer(headers="aeg-sas-key, aeg-sas-token, aeg-channel-name")
     add_general_regex_sanitizer(
         value="fakeresource",
-        regex="(?<=\\/\\/)[a-z-]+(?=\\.westus2-1\\.eventgrid\\.azure\\.net/api/events)",
+        regex="(?<=\\/\\/)[.*]+(?=\\.eastus-1\\.eventgrid\\.azure\\.net/api/events)",
     )
+
+    add_oauth_response_sanitizer()
+    add_header_regex_sanitizer(key="Set-Cookie", value="[set-cookie;]")
+    add_header_regex_sanitizer(key="Cookie", value="cookie;")
+
+    add_body_key_sanitizer(json_path="$..id", value="id")
+
+    client_id = os.getenv("AZURE_CLIENT_ID", "sanitized")
+    client_secret = os.getenv("AZURE_CLIENT_SECRET", "sanitized")
+    eventgrid_client_id = os.getenv("EVENTGRID_CLIENT_ID", "sanitized")
+    eventgrid_client_secret = os.getenv("EVENTGRID_CLIENT_SECRET", "sanitized")
+    tenant_id = os.getenv("AZURE_TENANT_ID", "sanitized")
+    eventgrid_topic_endpoint = os.getenv("EVENTGRID_TOPIC_ENDPOINT", "sanitized")
+    # Need to santize namespace for eventgrid_topic:
+    try:
+        eventgrid_hostname = (eventgrid_topic_endpoint.split("https://")[1]).split("/api")[0]
```

Comment:
nit: you can use [urlsplit](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlsplit) to get the hostname